src/reader/IReader.java

```diff
@@ -0,0 +1,20 @@
+package reader;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class IReader {
```

Comment:
I indicates for interface and it is name convention in C# not Java. Even if then you should use BReader but i do not recommend that.